Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -195,15 +195,21 @@ - (void)authorizationController:(ASAuthorizationController *)controller didCompl
             }
         }
     }
+    RetainPtr<ASPublicKeyCredentialClientData> clientData = adoptNS([allocASPublicKeyCredentialClientDataInstance() initWithChallenge:toNSData(options.challenge).get() origin:callerOrigin.toString()]);
     if (includePlatformRequest) {
-        auto request = adoptNS([[allocASAuthorizationPlatformPublicKeyCredentialProviderInstance() initWithRelyingPartyIdentifier:*options.rp.id] createCredentialRegistrationRequestWithChallenge:toNSData(options.challenge).get() name:options.user.name userID:toNSData(options.user.id).get()]);
+        RetainPtr<ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest> request = adoptNS([[allocASAuthorizationPlatformPublicKeyCredentialProviderInstance() initWithRelyingPartyIdentifier:*options.rp.id] createCredentialRegistrationRequestWithClientData:clientData.get() name:options.user.name userID:toNSData(options.user.id).get()]);
```

Comment:
Can't this stay as an `auto`?

Reply:
I do not want `auto` here because it masks smart pointers and make our life harder when doing code auditing for smart pointer adoption. That said, I think this should have been `RetainPtr`, not `RetainPtr<ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest>` given that it's 2024 :)